src/main/java/org/gradle/profiler/ScenarioContext.java

```diff
@@ -0,0 +1,28 @@
+package org.gradle.profiler;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import java.util.UUID;
+
+public class ScenarioContext {
+    private final UUID invocationId;
+    private final String scenarioName;
+
+    public static ScenarioContext from(InvocationSettings invocationSettings, ScenarioDefinition scenarioDefinition) {
+        return new ScenarioContext(invocationSettings.getInvocationId(), scenarioDefinition.getName());
+    };
+
+    @VisibleForTesting
+    public ScenarioContext(UUID invocationId, String scenarioName) {
+        this.invocationId = invocationId;
+        this.scenarioName = scenarioName;
+    }
+
+    public String getUniqueScenarioId() {
+        return String.format("_%s_%s", invocationId.toString().replaceAll("-", "_"), scenarioName);
```

Comment:
This id will be used as part of a Java identifier. I think it is dangerous to use the scenario name here as-is. I could run the same scenario twice in one Gradle profiler invocation. How about we use the index of the scenario here?

Reply:
We didn't have a different "unique text" for multiple scenario executions before either. We could make this change, but I don't want to do it in this PR.

Reply:
I am not worried about the text being unique, but about the text not being good for a method name. For example, in the `gradle/gradle` build we use the display name for the Gradle profiler scenario name: https://github.com/gradle/gradle/blob/14acd460ee399dd78fe3a6938754e5687203df90/subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/fixture/GradleProfilerBuildExperimentRunner.java#L168

This won't be possible any more, since it contains spaces and other characters and we won't be able to use it as part of a method name.